mautic/mautic
# Using git tag instead of branch for history.js as it caused ssh issue…

…s for GitPod otherwise



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

https://github.com/mautic/mautic/commit/a7f833bd1ba6fb6fe525cf8f245ce70d99bf6f0b broke the Mautic installation in GitPod with this error: 

```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/browserstate/history.js.git
npm ERR! 
npm ERR! Warning: Permanently added 'github.com,140.82.121.3' (ECDSA) to the list of known hosts.
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! exited with error code: 128

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/gitpod/.npm/_logs/2023-05-16T07_07_50_462Z-debug.log
Script npm ci handling the npm-ci event returned with error code 1
Script @npm-ci was called via post-install-cmd
```

Whatever I tried it ended up with similar error. Installing via https instead of ssh somehow doesn't work as NPM always use ssh even if I explicitly requested ssh like `npm install git+https://github.com/browserstate/history.js.git`

I couldn't find the library in npmjs.com. If you search for "browserstate" which is the vendor of this library you get no results. And I previously made a mistake of installing wrong library called the same. Finally, Google came to rescue when I was able to find the right one:

https://www.npmjs.com/package/historyjs

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. There are no errors in the browser's devtools console
3. Clicking through the UI works via AJAX instead of refreshes. Check that the blue bar at the top that is indicating that something is loading exists when you click through the UI.
